File: packages/silo-client-java/src/main/java/in/org/quicko/silo/client/collections/CollectionSummary.java
package in.org.quicko.silo.client.collections;

import com.fasterxml.jackson.databind.JsonNode;
import in.org.quicko.silo.client.transport.JsonValues;
import in.org.quicko.silo.client.transport.Timestamps;
import java.time.Instant;

/**
 * One collection as the listing answers it: no schema, which is the one thing a
 * list of collections never needs to draw.
 */
public record CollectionSummary(
    String id,
    String name,
    long entries,
    boolean requiresAuth,
    Instant createdAt,
    Instant updatedAt) {

  public static CollectionSummary fromWire(JsonNode payload) {
    return new CollectionSummary(
        JsonValues.text(payload, "id"),
        JsonValues.text(payload, "name"),
        payload.path("entries").asLong(0),
        payload.path("requires_auth").asBoolean(false),
        Timestamps.at(payload, "created_at"),
        Timestamps.at(payload, "updated_at"));
  }
}

File: packages/silo-client-dart/lib/src/collections/collection_summary.dart
import '../transport/timestamps.dart';

/// A collection as the listing answers it, without its schema.
final class CollectionSummary {
  const CollectionSummary({
    required this.id,
    required this.name,
    required this.entries,
    required this.requiresAuth,
    required this.createdAt,
    required this.updatedAt,
  });

  factory CollectionSummary.fromJson(Map<String, Object?> json) => CollectionSummary(
    id: json['id'] as String,
    name: json['name'] as String,
    entries: json['entries'] as int,
    requiresAuth: json['requires_auth'] as bool,
    createdAt: Timestamps.parse(json['created_at']),
    updatedAt: Timestamps.parse(json['updated_at']),
  );

  final String id;
  final String name;
  final int entries;
  final bool requiresAuth;
  final DateTime createdAt;
  final DateTime updatedAt;
}
